test/unit/mem/copy/src/BufSlicing.cpp

```diff
@@ -0,0 +1,543 @@
+        alpaka::dim::DimInt< 2u >,
+        Idx
+    >;
+
+    static auto getVec(
+        Idx const & elDimOne,
+        Idx const & elDimTwo,
+        Idx const & elDimThree
+    ) -> Vec
+    {
+        alpaka::ignore_unused( elDimThree );
+        return Vec(
+            elDimOne,
+            elDimTwo
+        );
+    }
+
+    template< typename Data >
+    static auto cmpCorrectlySlicedBuffer(
+        Vec extents,
+        Vec extentsSubView,
+        Vec offsets
+    ) -> std::vector< Data >
+    {
+        std::vector< Data >
+            resSliced = std::vector< Data >( extentsSubView.prod( ) );
+        for( Idx x = 0; x < extentsSubView[0]; ++x )
+        {
+            for( Idx y = 0; y < extentsSubView[1]; ++y )
+            {
+                resSliced[x * extentsSubView[1] + y] = static_cast<Data>(
+                    ( x + offsets[0] ) * extents[1] + offsets[1] + y);
+            }
+        }
+        return resSliced;
+    }
+};
+
+// 3D explicit specialisation
+template< typename Idx >
+struct DimHelper<
+    alpaka::dim::DimInt< 3u >,
+    Idx
+>
+{
+    using Vec = alpaka::vec::Vec<
+        alpaka::dim::DimInt< 3u >,
+        Idx
+    >;
+
+    static auto getVec(
+        Idx const & elDimOne,
+        Idx const & elDimTwo,
+        Idx const & elDimThree
+    ) -> Vec
+    {
+        return Vec(
+            elDimOne,
+            elDimTwo,
+            elDimThree
+        );
+    }
+
+    template< typename Data >
+    static auto cmpCorrectlySlicedBuffer(
+        Vec extents,
+        Vec extentsSubView,
+        Vec offsets
+    ) -> std::vector< Data >
+    {
+        std::vector< Data >
+            resSliced = std::vector< Data >( extentsSubView.prod( ) );
+        for( Idx x = 0; x < extentsSubView[0]; ++x )
+        {
+            for( Idx y = 0; y < extentsSubView[1]; ++y )
+            {
+                for( Idx z = 0; z < extentsSubView[2]; ++z )
+                {
+                    resSliced[x * extentsSubView[2] * extentsSubView[1] +
+                              extentsSubView[2] * y +
+                              z] = static_cast<Data>(
+                        ( x + offsets[0] ) * extents[2] * extents[1] +
+                        ( y + offsets[1] ) * extents[2] +
+                        z +
+                        offsets[2]);
+                }
+            }
+        }
+        return resSliced;
+    }
+};
+
+template<
+    typename Idx,
+    typename Dim,
+    typename Data
+>
+struct TestSlicing
+{
+    template< typename Acc >
+    auto operator()( ) -> void
+    {
+        constexpr Idx extentsDimOne = 13;
+        constexpr Idx extentsDimTwo = 7;
+        constexpr Idx extentsDimThree = 5;
+
+        constexpr Idx extentsSubViewDimOne = 5;
+        constexpr Idx extentsSubViewDimTwo = 3;
+        constexpr Idx extentsSubViewDimThree = 3;
+
+        constexpr Idx offsetDimOne = 3;
+        constexpr Idx offsetDimTwo = 2;
+        constexpr Idx offsetDimThree = 1;
```

Comment:
The original of this test is using `auto const offsetView(alpaka::vec::Vec<Dim, Idx>::all(static_cast<Idx>(1)))` but we could also improve this to get some more irregularity like it is done here.